x-pack/plugins/lens/public/visualizations/partition/suggestions.ts

```diff
@@ -40,19 +40,24 @@ function shouldReject({ table, keptLayerIds, state }: SuggestionRequest<PieVisua
 
 function getNewShape(
   groups: TableSuggestionColumn[],
-  subVisualizationId?: PieVisualizationState['shape']
+  subVisualizationId?: PieVisualizationState['shape'],
+  currentShape?: PieVisualizationState['shape']
 ) {
   if (subVisualizationId) {
     return subVisualizationId;
   }
 
+  if (currentShape) {
+    return currentShape;
+  }
+
   let newShape: PieVisualizationState['shape'] | undefined;
 
   if (groups.length !== 1 && !subVisualizationId) {
     newShape = PieChartTypes.PIE;
   }
 
-  return newShape ?? PieChartTypes.DONUT;
+  return newShape ?? PieChartTypes.PIE;
```

Comment:
`subVisualizationId` is always defined at this point, so you can remove that check on the `if`.

Looking at this can be further improved and return just `PieChartTypes.PIE` because newShape will never have other values and if undefined we still default to `PieChartTypes.PIE`.